modules/samples/connectedcup/component/plugin/src/main/java/org/coffeeking/connectedcup/plugin/impl/dao/impl/ConnectedCupDAOImpl.java

```diff
@@ -0,0 +1,242 @@
+					propertyMap.put(ConnectedCupConstants.DEVICE_PLUGIN_PROPERTY_REFRESH_TOKEN, resultSet.getString("REFRESH_TOKEN"));
+					iotDevice.setDeviceProperties(propertyMap);
+
+					if (log.isDebugEnabled()) {
+						log.debug("Connected Cup service " + iotDeviceId + " data has been fetched from " +
+						          "Connected Cup database.");
+					}
+				}
+	        } catch (SQLException e) {
+	            String msg = "Error occurred while fetching Connected Cup device : '" + iotDeviceId + "'";
+	            log.error(msg, e);
+	            throw new IotDeviceManagementDAOException(msg, e);
+	        } finally {
+	            IotDeviceManagementDAOUtil.cleanupResources(stmt, resultSet);
+	            ConnectedCupDAO.closeConnection();
+	        }
+
+	        return iotDevice;
+	    }
+
+	    @Override
+	    public boolean addIotDevice(IotDevice iotDevice)
+			    throws IotDeviceManagementDAOException {
+	        boolean status = false;
+	        Connection conn = null;
+	        PreparedStatement stmt = null;
+	        try {
+	            conn = ConnectedCupDAO.getConnection();
+		        String createDBQuery =
+				        "INSERT INTO CONNECTED_CUP_DEVICE(CONNECTED_CUP_DEVICE_ID, DEVICE_NAME, ACCESS_TOKEN, REFRESH_TOKEN) VALUES (?, ?, ?, ?)";
+
+		        stmt = conn.prepareStatement(createDBQuery);
+		        stmt.setString(1, iotDevice.getIotDeviceId());
+		        stmt.setString(2, iotDevice.getIotDeviceName());
+		        stmt.setString(3, iotDevice.getDeviceProperties().get(ConnectedCupConstants.DEVICE_PLUGIN_PROPERTY_ACCESS_TOKEN));
+		        stmt.setString(4, iotDevice.getDeviceProperties().get(ConnectedCupConstants.DEVICE_PLUGIN_PROPERTY_REFRESH_TOKEN));
+			
+				
+				int rows = stmt.executeUpdate();
+				if (rows > 0) {
+					status = true;
+					if (log.isDebugEnabled()) {
+						log.debug("Connected Cup device " + iotDevice.getIotDeviceId() + " data has been" +
+						          " added to the Connected Cup database.");
+					}
+				}
+	        } catch (SQLException e) {
+	            String msg = "Error occurred while adding the Connected Cup device '" +
+	                         iotDevice.getIotDeviceId() + "' to the Connected Cup db.";
+	            log.error(msg, e);
+	            throw new IotDeviceManagementDAOException(msg, e);
+	        } finally {
+	            IotDeviceManagementDAOUtil.cleanupResources(stmt, null);
+	        }
+	        return status;
+	    }
+
+	    @Override
+	    public boolean updateIotDevice(IotDevice iotDevice)
+			    throws IotDeviceManagementDAOException {
+	        boolean status = false;
+	        Connection conn = null;
+	        PreparedStatement stmt = null;
+	        try {
+	            conn = ConnectedCupDAO.getConnection();
+	            String updateDBQuery =
+						"UPDATE CONNECTED_CUP_DEVICE SET  DEVICE_NAME = ?, ACCESS_TOKEN=?, REFRESH_TOKEN=? WHERE CONNECTED_CUP_DEVICE_ID = ?";
```

Comment:
Is line size correct here?

Reply:
This file is removed since the device type structure is changed now.